### PR TITLE
feature(Custom Fields): Add custom fields

### PR DESCRIPTION
**Pull request by geolessel**
This is based off of PR #30 so a lot of the code from there is in here.

### CODE REVIEW

File: test/pco_api/people/list/share_test.exs [hunk 1]
```diff
@@ -0,0 +1,64 @@
+defmodule PcoApi.People.List.ShareTest do
```
Comment (credoci) [inline test/pco_api/people/list/share_test.exs]: Modules should have a @moduledoc tag.

File: lib/pco_api/client.ex [hunk 2]
```diff
@@ -27,7 +27,10 @@ defmodule PcoApi.Client do
       end
 
       defp process_response_body(body) do
-        body |> Poison.decode!
+        cond do
```
Comment (credoci) [inline lib/pco_api/client.ex]: Cond statements should contain at least two conditions besides `true`.

File: test/pco_api/people/list/rule/condition_test.exs [hunk 3]
```diff
@@ -0,0 +1,55 @@
+defmodule PcoApi.People.List.Rule.ConditionTest do
```
Comment (credoci) [inline test/pco_api/people/list/rule/condition_test.exs]: Modules should have a @moduledoc tag.

File: test/pco_api/people/person/field_datum_test.exs [hunk 4]
```diff
@@ -0,0 +1,122 @@
+defmodule PcoApi.People.Person.FieldDatumTest do
```
Comment (credoci) [inline test/pco_api/people/person/field_datum_test.exs]: Modules should have a @moduledoc tag.

File: test/pco_api/people/field_definition/field_option_test.exs [hunk 5]
```diff
@@ -0,0 +1,90 @@
+defmodule PcoApi.People.Person.FieldDefinition.FieldOptionTest do
```
Comment (credoci) [inline test/pco_api/people/field_definition/field_option_test.exs]: Modules should have a @moduledoc tag.

File: test/pco_api/query_test.exs [hunk 6]
```diff
@@ -46,6 +46,6 @@ defmodule PcoApi.QueryTest do
 
   test "after_record allows piping" do
     expected = [{"after", "100"}, {"per_page", "100"}]
-    assert Query.per_page(100) |> Query.after_record(100)
+    assert (Query.per_page(100) |> Query.after_record(100)) == expected
```
Comment (credoci) [inline test/pco_api/query_test.exs]: Pipe chain should start with a raw value.

File: lib/pco_api/people/person/field_datum.ex [hunk 7]
```diff
@@ -0,0 +1,18 @@
+defmodule PcoApi.People.Person.FieldDatum do
```
Comment (credoci) [inline lib/pco_api/people/person/field_datum.ex]: Modules should have a @moduledoc tag.

File: lib/pco_api/people/connected_people.ex [hunk 8]
```diff
@@ -0,0 +1,8 @@
+defmodule PcoApi.People.ConnectedPeople do
```
Comment (credoci) [inline lib/pco_api/people/connected_people.ex]: Modules should have a @moduledoc tag.

File: test/pco_api/people/field_definition_test.exs [hunk 9]
```diff
@@ -0,0 +1,91 @@
+defmodule PcoApi.People.Person.FieldDefinitionTest do
```
Comment (credoci) [inline test/pco_api/people/field_definition_test.exs]: Modules should have a @moduledoc tag.

File: lib/pco_api/people/list/rule/condition/result.ex [hunk 10]
```diff
@@ -0,0 +1,9 @@
+defmodule PcoApi.People.List.Rule.Condition.Result do
```
Comment (credoci) [inline lib/pco_api/people/list/rule/condition/result.ex]: Modules should have a @moduledoc tag.

File: lib/pco_api/people/field_definition/field_option.ex [hunk 11]
```diff
@@ -0,0 +1,14 @@
+defmodule PcoApi.People.FieldDefinition.FieldOption do
```
Comment (credoci) [inline lib/pco_api/people/field_definition/field_option.ex]: Modules should have a @moduledoc tag.

File: lib/pco_api/people/list/share.ex [hunk 12]
```diff
@@ -0,0 +1,10 @@
+defmodule PcoApi.People.List.Share do
```
Comment (credoci) [inline lib/pco_api/people/list/share.ex]: Modules should have a @moduledoc tag.

File: lib/pco_api/people/list/rule/result.ex [hunk 13]
```diff
@@ -0,0 +1,16 @@
+defmodule PcoApi.People.List.Rule.Result do
```
Comment (credoci) [inline lib/pco_api/people/list/rule/result.ex]: Modules should have a @moduledoc tag.

File: test/pco_api/people/list/rule/condition/result_test.exs [hunk 14]
```diff
@@ -0,0 +1,55 @@
+defmodule PcoApi.People.List.Rule.Condition.ResultTest do
```
Comment (credoci) [inline test/pco_api/people/list/rule/condition/result_test.exs]: Modules should have a @moduledoc tag.

File: test/pco_api/people/list/rule/result_test.exs [hunk 15]
```diff
@@ -0,0 +1,55 @@
+defmodule PcoApi.People.List.Rule.ResultTest do
```
Comment (credoci) [inline test/pco_api/people/list/rule/result_test.exs]: Modules should have a @moduledoc tag.

File: test/pco_api/people/connected_people_test.exs [hunk 16]
```diff
@@ -0,0 +1,48 @@
+defmodule PcoApi.People.ConnectedPeopleTest do
```
Comment (credoci) [inline test/pco_api/people/connected_people_test.exs]: Modules should have a @moduledoc tag.

File: lib/pco_api/people/field_definition.ex [hunk 17]
```diff
@@ -0,0 +1,14 @@
+defmodule PcoApi.People.FieldDefinition do
```
Comment (credoci) [inline lib/pco_api/people/field_definition.ex]: Modules should have a @moduledoc tag.

File: test/pco_api/query_test.exs [hunk 18]
```diff
@@ -30,8 +30,8 @@ defmodule PcoApi.QueryTest do
   end
 
   test "per_page allows piping" do
-    expected = [{"where[b]", "bb"}, {"per_page", "30"}]
-    assert Query.where(b: "bb") |> Query.per_page(30)
+    expected = [{"per_page", "30"}, {"where[b]", "bb"}]
+    assert (Query.where(b: "bb") |> Query.per_page(30)) == expected
```
Comment (credoci) [inline test/pco_api/query_test.exs]: Pipe chain should start with a raw value.

File: lib/pco_api/people/list/rule/condition.ex [hunk 19]
```diff
@@ -0,0 +1,9 @@
+defmodule PcoApi.People.List.Rule.Condition do
```
Comment (credoci) [inline lib/pco_api/people/list/rule/condition.ex]: Modules should have a @moduledoc tag.

**Comment by jessejanderson** [comment]
Can you rebase now or whatever so it's only the new stuff?

**Comment by geolessel** [comment]
@jessejanderson done.

**Comment by jessejanderson** [comment]
👍

**Pull request merged**